protos/sigstore_rekor_v2.proto

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.rekor.v2;
+
+import "google/api/field_behavior.proto";
+
+import "sigstore_common.proto";
+
+option go_package = "github.com/sigstore/protobuf-specs/gen/pb-go/rekor/v2";
+option java_package = "dev.sigstore.proto.rekor.v2";
+option java_multiple_files = true;
+option java_outer_classname = "RekorProto";
+option ruby_package = "Sigstore::Rekor::V2";
+
+// KindVersion contains the entry's kind and api version.
+message KindVersion {
+        // Kind is the type of entry being stored in the log.
+        // See here for a list: https://github.com/sigstore/rekor/tree/main/pkg/types
+        string kind = 1 [(google.api.field_behavior) = REQUIRED];
+        // The specific api version of the type.
+        string version = 2 [(google.api.field_behavior) = REQUIRED];
+}
+
+// The checkpoint MUST contain an origin string as a unique log identifier,
+// the tree size, and the root hash. It MAY also be followed by optional data,
+// and clients MUST NOT assume optional data. The checkpoint MUST also contain
+// a signature over the root hash (tree head). The checkpoint MAY contain additional
+// signatures, but the first SHOULD be the signature from the log. Checkpoint contents
+// are concatenated with newlines into a single string.
+// The checkpoint format is described in
+// https://github.com/transparency-dev/formats/blob/main/log/README.md
+// and https://github.com/C2SP/C2SP/blob/main/tlog-checkpoint.md.
+// An example implementation can be found in https://github.com/sigstore/rekor/blob/main/pkg/util/signed_note.go
+message Checkpoint {
+        string envelope = 1 [(google.api.field_behavior) = REQUIRED];
+}
+
```

Comment:
These messages `KindVersion` and `Checkpoint` are repeated, they can probably be moved to a new common packaged (or be directly re-used from the rekor.v1 package). I don't think package information is used for anything but code gen -- but I Could be wrong. Either way moving it would require anyone importing the new builds to rewrite a little code.